bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -125,12 +125,12 @@ data:
     {{- else if eq .Values.externalAccess.service.type "NodePort" }}
     {{- if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
     export EXTERNAL_ACCESS_HOST="${MY_POD_IP}"
+    {{- else if .Values.externalAccess.service.domain }}
+    export EXTERNAL_ACCESS_HOST={{ .Values.externalAccess.service.domain }}
     {{- else if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
```

Comment:
After a second review, I think the problem with your settings is this line. As $HOST_IP is an auto-calculated value, this if should use "and" instead of "or" as we are using in the previous lines. That way the logic will use the pod or host IP if autodiscover is enabled, the domain or the IPs if you set them or will obtain the info with curl. 

That way, no matter were you move this if block, it'll always work the same way. WDYT?

Reply:
There may be a situation where both NodePort and autodiscovery are enabled, and it is generally expected to use hostIP + NodePort as the access point. According to Kubernetes conventions, useHostIPs is not explicitly enabled. If this is modified, it may cause confusion because users may be used to accessing with "curl ip" instead of the expected hostIP when NodePort and autodiscovery are enabled without useHostIPs being explicitly enabled.